src/saml.ts

```diff
@@ -68,8 +68,8 @@ async function processValidlySignedPostRequestAsync(
       throw new Error("Missing SAML issuer");
     }
     const nameID = await self._getNameIdAsync(self, dom);
-    if (nameID) {
-      profile.nameID = nameID.value!;
+    if (nameID.value) {
```

Comment:
Why the change from using `nameID` to `nameID.value` here?

Reply:
`self._getNameIdAsync` always returns an object with `.value`, so, in reality, what we want to do is check if `.value` has a value. That also allows us to remove the non-null assertion in the following line.

It follows the pattern directly after it with `nameID.format`.